nodejs/src/index.js

```diff
@@ -7,34 +7,43 @@ var Compression = exports.Compression = {
 	Gzip: 1
 };
 
-exports.write = function (struct, compression, callback) {
-	switch (compression || Compression.None) {
-		case Compression.None:
-			serialize(struct, callback);
-			break;
-		case Compression.Gzip:
-			zip(struct, callback);
-			break;
-		default:
-			callback(new Error('Unable to understand the compression applied'));
-			break;
-	}
+exports.write = function (struct, compression, callback, noSettings) {
+    if (!noSettings) {
```

Comment:
I see.

The no compression is a bit tricky, now the read needs to know how the data was compressed in the first place, which wasn't necessary.

How about

```js
exports.write = function (struct, compression, callback) {}
// compression.Disable

exports.read = function (Model, rawData, callback) {}
exports.read = function (Model, rawData, compression, callback) {}
```

If there are 3 params nothing changes, if 4 the 3rd is the compression and the 4th the callback?